reward/src/lib.rs

```diff
@@ -0,0 +1,532 @@
+    let signer: Arc<_> = SignerMiddleware::new(provider, wallet).into();
+    let profile = profile::default()?;
+    let storage = profile::read_only(&profile)?;
+    let repo = git::Repository::open(Path::new("."))?;
+
+    let mut revwalk = repo.revwalk()?;
+    let range = options
+        .range
+        .or_else(|| {
+            if !repo.tag_names(Some("rad/reward")).ok()?.is_empty() {
+                return Some("rad/reward..".to_string());
+            }
+            None
+        })
+        .ok_or_else(|| anyhow!("Can't determine the range of commits"))?;
+    revwalk.push_range(&range)?;
+
+    // from identities in local monorepo build a map of: fp -> ens
+    let mut ssh_to_ens: HashMap<String, _> = HashMap::new();
+    for identity in identities::any::list(&storage)?.flatten() {
+        if let Person(person) = identity {
+            let peer_ids = person
+                .delegations()
+                .iter()
+                .map(|key| -> PeerId { key.to_owned().into() })
+                .collect::<Vec<PeerId>>();
+            if let Ok(Some(ens)) = person.payload().get_ext::<Ens>() {
+                for peer_id in peer_ids {
+                    let fp = keys::to_ssh_fingerprint(&peer_id)?;
+                    ssh_to_ens.insert(fp.to_string(), ens.clone());
+                }
+            }
+        }
+    }
+
+    term::headline("Retrieving commits...");
+    let mut table = term::Table::default();
+    table.push([
+        term::format::bold("Commit"),
+        term::format::bold("Signature"),
+        term::format::bold("ENS"),
+    ]);
+
+    let mut head_sha1 = None;
+    // show commits in provided range and collect all ens which have contributed
+    let all_ens = revwalk
+        .filter_map(|sha1| {
+            let sha1 = match sha1 {
+                Err(_) => return None,
+                Ok(sha1) => sha1,
+            };
+            if head_sha1.is_none() {
+                head_sha1 = Some(sha1);
+            }
+
+            let ens = {
+                let fp = git::commit_ssh_fingerprint(Path::new("."), &sha1.to_string()).ok()?;
+                if fp.is_none() {
+                    table.push([
+                        term::format::highlight(sha1),
+                        term::format::tertiary("missing"),
+                        term::format::secondary("missing"),
+                    ]);
+                    return None;
+                }
+                let fp = fp?;
+                let ens = ssh_to_ens.get(&fp);
+                table.push([
+                    term::format::highlight(sha1),
+                    term::format::tertiary(fp.clone()),
+                    term::format::secondary(
+                        ens.map(|ens| ens.name.to_owned())
+                            .unwrap_or_else(|| String::from("missing")),
+                    ),
+                ]);
+                ens
+            };
```

Comment:
Seems like this block/assignment is redundant?

Reply:
https://github.com/radicle-dev/radicle-cli/compare/f4007e459f8ce654f3ca58ce48dc1f9910838534..a664622f7600c99dc0622b776816712d0b049d45#diff-26d7e6053383a7793e880ae8b6ee5d2f7d7c23ccd5e88e43f44052374e9f0f58R226